tests/integration_tests/test_pytorch_lightning.py

```diff
@@ -20,30 +23,42 @@ def __init__(self) -> None:
         super().__init__()
         self._model = nn.Sequential(nn.Linear(4, 8))
 
-    def forward(self, data: torch.Tensor) -> torch.Tensor:
+    def forward(self, data: torch.Tensor) -> torch.Tensor:  # type: ignore
 
         return self._model(data)
 
-    def training_step(self, batch: List[torch.Tensor], batch_nb: int) -> Dict[str, torch.Tensor]:
+    def training_step(  # type: ignore
+        self, batch: List[torch.Tensor], batch_nb: int
+    ) -> Dict[str, torch.Tensor]:
 
         data, target = batch
         output = self.forward(data)
         loss = F.nll_loss(output, target)
         return {"loss": loss}
 
-    def validation_step(self, batch: List[torch.Tensor], batch_nb: int) -> Dict[str, torch.Tensor]:
+    def validation_step(  # type: ignore
+        self, batch: List[torch.Tensor], batch_nb: int
+    ) -> Dict[str, torch.Tensor]:
 
         data, target = batch
         output = self.forward(data)
         pred = output.argmax(dim=1, keepdim=True)
         accuracy = pred.eq(target.view_as(pred)).double().mean()
         return {"validation_accuracy": accuracy}
 
-    def validation_epoch_end(self, outputs: List[Dict[str, torch.Tensor]]) -> None:
+    def validation_epoch_end(
+        self,
+        outputs: Union[
+            List[Union[torch.Tensor, Dict[str, Any]]],
+            List[List[Union[torch.Tensor, Dict[str, Any]]]],
+        ],
+    ) -> None:
         if not len(outputs):
             return
 
-        accuracy = sum(x["validation_accuracy"] for x in outputs) / len(outputs)
+        accuracy = sum(
+            x["validation_accuracy"] for x in cast(List[Dict[str, torch.Tensor]], outputs)
+        ) / len(outputs)
```

Comment:
[Notes] `outputs` can be `List[torch.Tensor]`, but original code apparently expects `List[Dict[str, torch.Tensor]]` , so this fix is acceptable to me.